diff -up fish-1.23.0/exec.c.BAD fish-1.23.0/exec.c
--- fish-1.23.0/exec.c.BAD      2008-09-15 16:28:52.000000000 -0400
+++ fish-1.23.0/exec.c  2008-09-15 16:29:56.000000000 -0400
@@ -24,6 +24,14 @@
 #include <dirent.h>
 #include <time.h>

+#if defined(_SC_ARG_MAX)
+# if defined(ARG_MAX)
+#  undef ARG_MAX
+# endif
+# define ARG_MAX sysconf (_SC_ARG_MAX)
+#endif
+
+
 #ifdef HAVE_SIGINFO_H
 #include <siginfo.h>
 #endif
